### PR TITLE
chore: Updating Go examples

### PR DESCRIPTION
**Pull request by lxfontes**
Refactored Go component examples code from wasi-preview1 to wasi-preview2.
CI Changes:
- Not building Go components with wash 0.29 as it doesn't support wasipreview2 for tinygo
- Bumped Go version to 1.23 in examples
- Bumped TinyGo version to 0.33 in examples
- Added `wasm-tools` via binstall ( wit-bindgen-go & tiny-go requirement )
- Added examples entry for `http-client-tinygo` ( was missing )

### CODE REVIEW

File: examples/golang/components/http-echo-tinygo/wasmcloud.toml [hunk 1]
```diff
@@ -24,4 +24,4 @@ claims = [
   "wasmcloud:builtin:logging",
 ]
 wit_world = "echo"
-wasm_target = "wasm32-wasi"
+wasm_target = "wasm32-wasi-preview2"
```
Comment (ricochet) [inline examples/golang/components/http-echo-tinygo/wasmcloud.toml:27]: target should be wasm32-wasi-wasip2

**Comment by joonas** [comment]
@lxfontes did you mean to delete [`examples/golang/components/http-client-tinygo/local.wadm.yaml`](https://github.com/wasmCloud/wasmCloud/pull/3132/files#diff-f2b024374d231e46476ec5e8797de2b85026ca019a5cc427b0b06b0cdb74da0f) as part of this?

**Review by ricochet — APPROVED** [review]

File: examples/golang/components/http-echo-tinygo/wasmcloud.toml (hunk 1 above)
Comment (ricochet) [reply examples/golang/components/http-echo-tinygo/wasmcloud.toml:27]: Actually realizing we haven't updated everywhere in wasmCloud to only refer to wasm32-wasip2 so OK if not fixed in this PR. Ref [rust wasi-targets](https://blog.rust-lang.org/2024/04/09/updates-to-rusts-wasi-targets.html).

File: examples/golang/components/http-client-tinygo/wit/world.wit [hunk 2]
```diff
@@ -1,6 +1,9 @@
 package wasmcloud:httpclient;
 
 world hello {
+  // tinygo
```
Comment (brooksmtownsend) [inline examples/golang/components/http-client-tinygo/wit/world.wit:4]: nit: did TinyGo need this? or was this just a leftover comment

**Review by brooksmtownsend — COMMENTED** [review]
Quick question, but this looks _awesome_! Not approving explicitly quite yet since this changes the code & we want to make sure we update the TinyGo quickstart on wasmcloud.com

File: examples/golang/components/http-client-tinygo/wit/world.wit (hunk 2 above)
Comment (lxfontes) [reply examples/golang/components/http-client-tinygo/wit/world.wit:4]: unfortunately tinygo requires that `import` to be present, and was looking for a way to express "hey, this is tinygo specific"

**Comment by lxfontes** [comment]
closing this as we drifted a loooooooot

**Pull request closed without merging**